lix/packages/sdk/src/discussion/start-discussion.ts

```diff
@@ -0,0 +1,68 @@
+import { Kysely } from "kysely";
+import type { LixDatabaseSchema } from "../database/schema.js";
+import { v4 } from "uuid";
+
+export async function startDiscussion(args: {
+	db: Kysely<LixDatabaseSchema>;
+	currentAuthor: string;
+	changes: { id: string }[];
```

Comment:
I would use `changeIds` as properties instead of an array of ids. Will make the API more ergonomical because the change object doesn't need to be pruned. 

```ts 
await createChangeDiscussion({ 
  // ...
  changeIds: changes.map((c) => c.id)
})
```

Else, a new object needs to be created for every change.

```ts
await createChangeDiscussion({ 
  // ...
  changes: changes.map((c) => ({ c.id })) 
})

```

Reply:
EDIT: due to https://github.com/opral/monorepo/pull/3121#discussion_r1757318548, i would have the following signature 

```ts
await createDiscussion(args: { 
   // ...
  changeIds?: Array<string>
})
```

Making change ids optional means that we can add other entity maps in the future. 

```diff
await createDiscussion(args: { 
   // ...
  changeIds?: Array<string>
+ branchIds?: Array<string>
})
```

Reply:
The idea was the other way around - we most likely have an array of changes (as in your example) in the application code already - no need to call map just pass the changes. I could change the type to `Pick<Change, 'id'>` to clarify this.

The function internally maps the ids anyway we could also make the type a mixed type:

Like Pick<Change, 'id'>[] | Change["id"][] - I am ambiguous as this type :D

Reply:
Function overloads are hard to discover. I see where you are coming from, but the assumption is that changes have been queried. 

```ts
const changes = await lix.db.selectFrom("change")

await createDiscussion({ changes })
```

The ugly mapping comes into play when a table that only exposes the change_id is queried. 

```ts
const other = await lix.db.selectFrom("otherTable")

await createDiscussion({ changes: other.map((o) => ({ id: o.change_id}) })
```

I would go with `change_id` :)

Reply:
> Function overloads are hard to discover. 

so we won't use it here?!

> I would go with change_id :)

Not 100 % which one you preferred now:

```
await createDiscussion(args: { 
   // ...
  changeIds?: Array<string>
})
```

or 

```
await createDiscussion(args: { 
   // ...
  changes?: {change_id: string}[]
})
```

Reply:
await createDiscussion(args: { 
   // ...
  changeIds?: Array<string>
})

this one